From ee8ec51cd5f3b3e10704e87dc813ae117dbb37bb Mon Sep 17 00:00:00 2001
From: Hu Bingquan <bingquan.hu@intel.com>
Date: Wed, 20 Apr 2016 14:05:45 +0800
Subject: [PATCH 4/4] input: silead touch: reset before suspending

This is a WA from vendor to fix the i2c register writing failure.

Change-Id: Ib5a102a11d0cf1b0922ceed05095070590074fc4
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-29185
Signed-off-by: Hu Bingquan <bingquan.hu@intel.com>
Reviewed-on: https://android.intel.com:443/496878
---
 drivers/input/touchscreen/silead.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/input/touchscreen/silead.c b/drivers/input/touchscreen/silead.c
index 3bdf036..4187707 100644
--- a/drivers/input/touchscreen/silead.c
+++ b/drivers/input/touchscreen/silead.c
@@ -188,6 +188,7 @@ static void silead_ts_set_power(struct i2c_client *client,
 					SILEAD_CLOCK);
 		}
 		else {
+			i2c_smbus_write_byte_data(client, SILEAD_REG_RESET,	SILEAD_CMD_RESET);
 			ret = i2c_smbus_write_byte_data(client, SILEAD_REG_CLOCK,
 					SILEAD_CLOCK_OFF);
 		}
-- 
1.9.1

